### PR TITLE
[#6587] IPFS hosted apps should not share cookies/etc

### LINKED ISSUE #6587: IPFS hosted apps should not share cookies/etc
### User Story

As a user, I want to know that my data is not shared between different IPFS hosted apps (cookies, localStorage). As we are currently leveraging infura subdomains 

### Description

*Type*: Bug

*Summary*: Currently opening an ENS/IPFS app will redirect to `ipfs.infura.io/..`. Because of web cookies and localStorage policues this leads to share data between DApps. Make sure this type of sharing doesn't happen.

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #6587

encode ipfs hash in base32 and redirect to base32hash.infura.status.im

related changes in infra 
https://github.com/status-im/infra-misc/commit/5185fc87bc2878dd4f707408ea904a3827586d75
https://github.com/status-im/infra-misc/commit/348cf2441afa99aa6a9b5c23db63f414c0af6661

### CODE REVIEW

File: src/status_im/browser/core.cljs [hunk 1]
```diff
@@ -177,14 +178,19 @@
   [{:keys [db] :as cofx} proto-code hash]
   (let [current-url (get-current-url (get-current-browser db))
         host (http/url-host current-url)
-        gateways (if (= swarm-proto-code proto-code) "swarm-gateways.net/bzz:/" "ipfs.infura.io/ipfs/")]
+        path  (subs current-url (+ (.indexOf current-url host) (count host)))
+        gateway (if (= ipfs-proto-code proto-code)
+                  (let [base32hash (-> (.encode js-dependencies/hi-base32 (alphabase.base58/decode hash))
```
Comment (jeluard) [inline src/status_im/browser/core.cljs:183]: Extract as a separate function?

**Review by jeluard — APPROVED** [review]

**Review by jakubgs — APPROVED** [review]
I skimmed the code of [hi-base32](https://github.com/emn178/hi-base32/blob/master/src/base32.js) package for any red flags and couldn't see anything dodgy. But I did not audit the algorithm for correctness.

**Comment by asemiankevich** [comment]
@flexsurfer is this expected?
image

**Comment by flexsurfer** [comment]
@asemiankevich no :)

**Comment by flexsurfer** [comment]
@asemiankevich we found and fixed a small problem with the certificate in our infra, idk if it was the reason, can't reproduce it, so if you can still reproduce please provide steps , thanks

**Comment by asemiankevich** [comment]
@flexsurfer could you please rebase?

**Comment by flexsurfer** [comment]
@asemiankevich done

**Comment by yenda** [comment]
This never built for desktop why did it got merged?

**Comment by jakubgs** [comment]
Good question. @flexsurfer we probably should revert this merge.

**Comment by flexsurfer** [comment]
@yenda @jakubgs good question, i'm working on the fix

**Comment by flexsurfer** [comment]
@jakubgs interesting why it failed only for desktop builds, do we run tests only in desktop builds?

**Comment by yenda** [comment]
@antdanchenko @flexsurfer I installed this build on my phone and I have the 404 as well

**Comment by flexsurfer** [comment]
@yenda @antdanchenko fixed in our infra, sorry for inconvenience

**Pull request merged**